elastic/integrations
# [logtash] align logstash logs ingestion

### Summary

Closes https://github.com/elastic/integrations/issues/4048

- Added pipeline and static tests to the logstash logs
- Added log4j2 configuration to output to file instead of stdout and logstash configuration to trigger slowlogs
- Updated saved search filters to look for `data_stream.dataset` property instead of the outdated prefix filter. Since dashboard and visualizations rely on the saved search it is the only asset that was updated functionally, but I also re-generated the other ones so we have them in the latest format

The service is set to output the logs in json, I've also verified that plaintext logs were successfully ingested and added test cases for both json and plain. You can comment [this line](https://github.com/elastic/integrations/blob/d11139b00109489dc15fa94208b13653006295a7/packages/logstash/_dev/deploy/docker/config/logstash.yml#L4) to test plaintext.

#### Notes
- When building the package we get the following warning, does this mean that references are deprecated ?
  > Warning: references found in dashboard kibana/dashboard/logstash-Logs-Logstash-Log-Dashboard.json: logstash-0b1dace0-cbdb-11e7-9852-73e0a9df1bb6 (visualization), logstash-e90b7240-cbda-11e7-9852-73e0a9df1bb6 (visualization)
- The dashboards look for fields that don't exist or are renamed. We'll need input from logstash team to know whether we need to replace or remove them

### Testing
- build logstash package `(cd packages/logstash && elastic-package build -v)`
- start elastic stack `elastic-package stack up -v -d --version 8.5.0-SNAPSHOT`
- start logstash service `(cd packages/logstash && elastic-package service up -v)`
- launch kibana at https://localhost:5601 and add the logstash integration. The logs are located at `/tmp/service_logs/logstash/logstash-json.log` and `/tmp/service_logs/logstash/logstash-slowlog-json.log`.
- navigate to Discover and verify both `logstash.log` and `logstash.slowlog` datasets are ingested
- navigate to the Logstash integration Assets and verify every asset is working

<img width="1222" alt="Screenshot 2022-09-14 at 22 43 09" src="https://user-images.githubusercontent.com/5239883/190258438-1e0376b8-76ea-42a1-b820-f0b2f624e95d.png">
